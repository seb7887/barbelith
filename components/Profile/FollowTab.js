import Link from 'next/link';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import withStyles from '@material-ui/core/styles/withStyles';

const FollowTab = ({ classes, users }) => (
  <div className={classes.root} data-testid='follow-tab'>
    <GridList cellHeight={160} className={classes.gridList} cols={4}>
      {users.map(user => (
        <GridListTile style={{ height: 120 }} key={user._id}>
          <Link href={`/profile/${user._id}`} prefetch>
            <a>
              <Avatar src={user.avatar} className={classes.bigAvatar} />
              <Typography
                component='h3'
                variant='subtitle1'
                className={classes.tileText}
              >
                {user.name}
              </Typography>
            </a>
          </Link>
        </GridListTile>
      ))}
    </GridList>
  </div>
)

const styles = theme => ({
  root: {
    marginTop: theme.spacing.unit * 2,
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden'
  },
  bigAvatar: {
    width: 60,
    height: 60,
    margin: 'auto'
  },
  gridList: {
    width: 300,
    [theme.breakpoints.up('sm')]: {
      width: 400
    }
  },
  tileText: {
    textAlign: 'center',
    marginTop: 10,
    textDecoration: 'none',
    color: theme.palette.secondary.main
  }
});

export default withStyles(styles)(FollowTab);