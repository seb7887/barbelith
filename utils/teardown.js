use barbelith-db;
show collections;
db.users.remove({});
db.sessions.remove({});
db.posts.remove({});
