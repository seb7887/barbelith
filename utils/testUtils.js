import faker from 'faker';

export const validUser = () => ({
  name: 'kingmob',
  email: 'king@mob.com',
  password: 'km1234'
});

export const fakeUser = () => ({
  name: faker.internet.userName(),
  email: faker.internet.email(),
  password: faker.internet.password()
});

export const mockedUser = () => ({
  _id: '7',
  avatar: '/static/images/avatar.jpg',
  createdAt: '2019-02-1234',
  email: faker.internet.email(),
  followers: ['1', '2', '3'],
  following: ['3'],
  name: 'Test',
  updatedAt: '2019-02-1234'
});

export const updatedUser = () => ({
  _id: '7',
  name: 'kingmob',
  email: faker.internet.email(),
  avatar: faker.image.avatar(),
  about: faker.lorem.text()
});

export const generateUsers = quantity => {
  const generatedUsers = [];

  for (let i = 0; i < quantity; i++) {
    const user = {
      _id: `${i}`,
      name: `test${i}`,
      avatar: faker.image.avatar()
    };
    generatedUsers.push(user);
  }

  return generatedUsers;
};

export const generateComments = quantity => {
  const generatedComments = [];

  for (let i = 0; i < quantity; i++) {
    const comment = {
      _id: faker.random.uuid().toString(),
      postedBy: {
        _id: '7',
        name: 'test',
        avatar: faker.image.avatar()
      },
      text: faker.lorem.text()
    };
    generatedComments.push(comment);
  }
  return generatedComments;
};

export const fakePost = () => ({
  _id: '7',
  postedBy: {
    _id: '7',
    name: 'test',
    avatar: faker.image.avatar()
  },
  createdAt: faker.date.recent().toString(),
  text: faker.lorem.text(),
  image: faker.image.avatar(),
  likes: ['7'],
  comments: []
});
