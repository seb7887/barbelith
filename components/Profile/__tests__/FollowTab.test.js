/* eslint-env jest */
import 'react-testing-library/cleanup-after-each';

import { render } from 'react-testing-library';
import { generateUsers } from '../../../utils/testUtils';
import FollowTab from '../FollowTab';

const user = generateUsers(1);

jest.mock('next/link', () => {
  return ({ children }) => {
    return children;
  }
});

describe('<FollowTab/>', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <FollowTab users={[]} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('show users correctly', () => {
    const { getByTestId } = render(
      <FollowTab users={user} />
    );

    expect(getByTestId('follow-tab').textContent).toContain(user[0].name);
  });
});
