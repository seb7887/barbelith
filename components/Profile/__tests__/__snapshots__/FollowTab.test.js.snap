// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowTab/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="FollowTab-root-1"
    data-testid="follow-tab"
  >
    <ul
      class="MuiGridList-root-5 FollowTab-gridList-3"
      style="margin: -2px;"
    />
  </div>
</DocumentFragment>
`;
